### ISSUE TITLE
[valid-expect-in-promise] Enforce for Observable .subscribe() also

### ISSUE THREAD

**Issue by ericrav** [jest-community/eslint-plugin-jest#272]
If you use an RxJS Observable and put an `expect` in the `subscribe` callback, you run into the same issue as with promises:

```js
it('passes', () => {
  someObservable.subscribe(() => {
    expect(false).toBe(true);
  });
});
```

Can `valid-expect-in-promise` be extended or duplicated to check for `subscribe`?

**Comment by SimenB** [comment]
I like the idea!
It might make more sense as a separate rule? Not sure, maybe not :)

**Comment by ericrav** [comment]
You're right, it should be a separate rule because the fix is different. You can't return the Observable subscription—you can only either call `done()` or convert it to a promise with `observable.toPromise()`

**Comment by SimenB** [comment]
Right, thanks.

PR welcome! 😀

**Comment by G-Rath** [comment]
Is this still a thing we want? I feel like `done` callbacks have fallen out of favor (`valid-expect-in-promise` completely bails if it finds one), and you've said you can convert the observable to a promise using `toPromise` which should mean that `valid-expect-in-promise` will work on the result just fine?

I'm not against the rule per say, but not sure if it belongs with us currently since it's using a third-party library that (afaik) is very unrelated to Jest and I've not seen any mention of it or similar patterns in other issues.

**Comment by ericrav** [comment]
Good points. My team no longer has a need for this & no one else has commented, so I'll close the issue
